game-app/game-core/src/main/java/games/strategy/triplea/ui/TechnologyPanel.java

```diff
@@ -0,0 +1,288 @@
+    techTable.getColumnModel().getColumn(0).setPreferredWidth(500);
+    // show icons for players:
+    final TableCellRenderer componentRenderer = new JComponentTableCellRenderer();
+    for (int i = 1; i < techTable.getColumnCount(); i++) {
+      final TableColumn column = techTable.getColumnModel().getColumn(i);
+      column.setHeaderRenderer(componentRenderer);
+      final String player = techTable.getColumnName(i);
+      final JLabel value = new JLabel("", getIcon(player), SwingConstants.CENTER);
+      value.setToolTipText(player);
+      column.setHeaderValue(value);
+    }
+    // show tooltip for technology names
+    final TableCellRenderer techNameComponentRenderer = new TechNameComponentRenderer();
+    final TableColumn techNameColumn = techTable.getColumnModel().getColumn(0);
+    techNameColumn.setHeaderRenderer(techNameComponentRenderer);
+    techNameColumn.setCellRenderer(techNameComponentRenderer);
+    return techTable;
+  }
+
+  static final class TechNameComponentRenderer extends DefaultTableCellRenderer {
+    @Override
+    public void setValue(Object aValue) {
+      setToolTipText(aValue.toString());
+      super.setValue(aValue);
+    }
+  }
+
+  public void setGameData(final GameData data) {
+    gameData.removeDataChangeListener(this);
+    gameData = data;
+    gameData.addDataChangeListener(this);
+    gameDataChanged(null);
+  }
+
+  @Override
+  public void gameDataChanged(final Change change) {
+    techModel.markDirty();
+    SwingUtilities.invokeLater(this::repaint);
+  }
+
+  /**
+   * Gets the small flag for a given PlayerId.
+   *
+   * @param player the player to get the flag for
+   * @return ImageIcon small flag
+   */
+  protected ImageIcon getIcon(final GamePlayer player) {
+    ImageIcon icon = mapPlayerImage.get(player);
+    if (icon == null && uiContext != null) {
+      final Image img = uiContext.getFlagImageFactory().getSmallFlag(player);
+      icon = new ImageIcon(img);
+      icon.setDescription(player.getName());
+      mapPlayerImage.put(player, icon);
+    }
+    return icon;
+  }
+
+  protected @Nullable ImageIcon getIcon(final String playerName) {
+    final GamePlayer player = gameData.getPlayerList().getPlayerId(playerName);
+    if (player == null) {
+      return null;
+    }
+    return getIcon(player);
+  }
+
+  private void fillPlayerIcons() {
+    for (final GamePlayer p : gameData.getPlayerList().getPlayers()) {
+      getIcon(p);
+    }
+  }
+
+  static class JComponentTableCellRenderer implements TableCellRenderer {
+    @Override
+    public Component getTableCellRendererComponent(
+        final JTable table,
+        final Object value,
+        final boolean isSelected,
+        final boolean hasFocus,
+        final int row,
+        final int column) {
+      return (JComponent) value;
+    }
+  }
+
+  class TechTableModel extends AbstractTableModel {
+    private static final long serialVersionUID = -4612476336419396081L;
```

Comment:
Must be unique, but is duplicate to constant from StatPanel.java (class TechTableModel)
Is there anything that really has to be stored/loaded on game save/load.